horreum-java-client/src/test/java/io/hyperfoil/tools/HorreumClientTest.java

```diff
@@ -0,0 +1,77 @@
+package io.hyperfoil.tools;
+
+import io.hyperfoil.tools.horreum.api.RunService;
+import io.hyperfoil.tools.horreum.entity.json.Access;
+import io.hyperfoil.tools.yaup.json.Json;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import javax.ws.rs.BadRequestException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.Instant;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class HorreumClientTest extends HorreumTestBase {
+
+    @Test
+    @Order(1)
+    public void JsonPayload() {
+
+        Instant stop = Instant.now();
+        Instant start = stop.minusSeconds(60 * 5);
+
+        Json payload = new Json();
+        payload.add("id", "1");
+        payload.add("start", start.toString());
+        payload.add("stop", stop.toString());
+        payload.add("build-hash", "965c96d8d21df38de4e0907efb5437b2ee462088");
+
+        io.hyperfoil.tools.horreum.entity.json.Test test = getExistingtest();
+
+        assertNotNull(test);
+        assertEquals(getProperty("horreum.test.description"), test.description);
+
+        RunService.RunCounts runCounts = horreumClient.runService.runCount(test.id);
+        assertEquals(0, runCounts.total);
+
+        try {
+            horreumClient.runService.addRunFromData("$.start", "$.stop", test.name, getProperty("horreum.test.owner"), Access.PUBLIC, null, null, "test", payload);
+        } catch (BadRequestException badRequestException) {
+            fail(badRequestException.getMessage() + (badRequestException.getCause() != null ? " : " + badRequestException.getCause().getMessage() : ""));
```

Comment:
this was here to allow me to see what the cause of the bad request was. for example: https://github.com/Hyperfoil/Horreum/blob/master/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java#L319

If there is an exception thrown in the server (e.g. "Cannot identify test name.", "Failed to find test "  etc) this is encoded in the payload of the response.  If I leave the test framework to handle the exception, it does not print out that payload and I have no clue as to **why** it was a bad request. This may be due to the way ServiceException is building the response (https://github.com/Hyperfoil/Horreum/blob/fbccd421931d65aa19a9131103e83458b01e6d88/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/ServiceException.java#L26), but I did not look at validating that behaviour as part of this PR